.github/workflows/ledger-suite-release.yml

```
@@ -0,0 +1,647 @@
+                  // the first matching release we find is the latest one
+                  const latestPrevious = matchingReleases[0];
+                  prevTag = latestPrevious.tag_name;
+                  foundMatch = true;
+
+                  console.log(`Found latest previous release: ${prevTag}`);
+                  break;
+                }
+
+                page++;
+              }
+
+              if (page > maxPages) {
+                console.log(`Warning: Reached maximum page limit (${maxPages}) - stopping pagination`);
+              }
+
+              if (prevTag) {
+                // Validate the release exists and is accessible
+                console.log(`Validating previous release: ${prevTag}`);
+                try {
+                  const releaseCheck = await github.rest.repos.getReleaseByTag({
+                    owner: context.repo.owner,
+                    repo: context.repo.repo,
+                    tag: prevTag
+                  });
+
+                  console.log(`Previous release validated: ${prevTag}`);
+                  prevTagValid = true;
+                } catch (error) {
+                  if (error.status === 404) {
+                    console.log(`Warning: Previous release tag ${prevTag} exists but release not found`);
+                    prevTag = '';
+                    prevTagValid = false;
+                  } else {
+                    throw error;
+                  }
+                }
+              } else {
+                console.log(`No previous release found for type: ${releaseType}`);
+              }
+
+              core.setOutput('prev_tag', prevTag);
+              core.setOutput('prev_tag_valid', prevTagValid.toString());
+
+            } catch (error) {
+              console.error(`Error fetching releases: ${error.message}`);
+              // Set empty outputs on error
+              core.setOutput('prev_tag', '');
+              core.setOutput('prev_tag_valid', 'false');
+              throw error;
+            }
+
+      - name: Generate git log
+        id: git_log
+        run: |
+          # github-script does not support formatting for the `git log` command, so this step is in shell script.
+
+          # Only use previous tag if it exists and is valid
+          if [[ -n "${{ steps.prev_tag.outputs.prev_tag }}" && "${{ steps.prev_tag.outputs.prev_tag_valid }}" == "true" ]]; then
+            LOG_RANGE="${{ steps.prev_tag.outputs.prev_tag }}..${{ steps.parse_tag.outputs.tag_name }}"
+            GIT_COMMAND="git log --format=\"%C(auto) %h %s\" $LOG_RANGE -- ${{ steps.variables.outputs.log_paths_string }}"
+            echo "Generating git log from ${{ steps.prev_tag.outputs.prev_tag }} to ${{ steps.parse_tag.outputs.tag_name }}"
+          else
+            # If no previous tag or invalid previous tag, get all commits
+            LOG_RANGE=""
+            GIT_COMMAND="git log --format=\"%C(auto) %h %s\" -- ${{ steps.variables.outputs.log_paths_string }}"
```

Quote the variable to prevent potential command injection.
```suggestion
            GIT_COMMAND="git log --format=\"%C(auto) %h %s\" -- \"${{ steps.variables.outputs.log_paths_string }}\""
```